splink/input_column.py

```diff
@@ -1,112 +1,153 @@
 
+    def add_suffix(self, suffix: str):
+        return replace(self, name=self.name + suffix) if suffix else self
 
-def add_prefix(tree, prefix) -> Expression:
-    tree = tree.copy()
-    identifier_string = tree.find(exp.Identifier).this
-    identifier_string = f"{prefix}{identifier_string}"
-    tree.find(exp.Identifier).args["this"] = identifier_string
-    return tree
+    def add_table(self, table: str):
+        return replace(self, table=table) if table else self
 
+    def add_alias(self, alias: str) -> exp.Alias:
+        """Alias expects a SQLglot Identifier class or a str.
 
-def add_table(tree, tablename) -> Expression:
-    tree = tree.copy()
-    table_identifier = exp.Identifier(this=tablename, quoted=True)
-    identifier = tree.find(exp.Column)
-    identifier.args["table"] = table_identifier
-    return tree
+        For safety, only pass strings to ensure we capture the entire
+        column reference, not just the identifier.
 
+        For example, col['lat'] is made up of both an identifier and bracket
+        index; the latter of which would be lost if passing identifiers.
+        """
+        # The identifier should be quoted or unquoted before being passed.
+        # This is to ensure we get the correct output.
+        alias = exp.Identifier(this=alias, quoted=self.quoted)
+        return self.build_column_tree().as_(alias)
 
-def remove_quotes_from_identifiers(tree) -> Expression:
-    tree = tree.copy()
-    for identifier in tree.find_all(exp.Identifier):
-        identifier.args["quoted"] = False
-    return tree
+    def add_column_reference(self, index_name: str | int):
+        """
+        Manually add a column reference to your column - ['idx'], for example.
+
+        This is useful during the parsing process or when you try to combine
+        illegal name and bracket combinations such as 'SUR NAME['lat']',
+        without needing to add quotes to the column identifier.
+        """
+        is_str = type(index_name) == str
+        self.column_reference = exp.Literal(this=index_name, is_string=is_str)
+        return self
+
+    def build_column_tree(self):
+        # Create our column representation
+        column = sqlglot.column(col=self.name, table=self.table, quoted=self.quoted)
+        # Column is made up of the identifier + bracket index (if it exists)
+        if self.column_reference:
+            return exp.Bracket(this=column, expressions=[self.column_reference])
+        else:
+            return column
+
+    @classmethod
+    def from_sqlglot_column(
+        cls, column_tree: exp.Column, column_reference: exp.Identifier = None
+    ):
+        return cls(column_tree.name, column_reference=column_reference)
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}("
+            f"name='{self.name}', quoted={self.quoted}, "
+            f"column_reference={self.column_reference.sql()}"
+        )
 
 
 class InputColumn:
```

Comment:
How do we describe this clearly in a way which distinguishes it from the `ColumnTreeBuilder`?

Reply:
I've tweaked the docstring for this.

In its current form, `InputColumn` is primarily an interface class (automating the building process). I've tried to reflect this in the updated docstring.